provider/utils.go

```
@@ -48,7 +48,7 @@ func parseChallengeFromRequest(r *http.Request, key string) (string, error) {
 	return challenge, nil
 }
 
-func putAndRedirect(url string, body interface{}, w http.ResponseWriter, r *http.Request, client *http.Client) {
+func putAccept(url string, body interface{}, client *http.Client) string {
```

this function should return an error (you swallowed all errors, including the http error, which looks dangerous)

---

also, I'm not sure it's a very good idea to factorize this part of code, body is an interface because it might be either  a BodyAcceptOAuth2Login or a BodyAcceptOAuth2Consent and it's a bit confusing.
I'd rather duplicate a few lines of code and have less abstraction, what do you think ?